File: Programmers/Lv2/Lv2_기능개발.py
def solution(progresses, speeds):
    answer = []
    temp = []
    for i, j in zip(progresses, speeds):
        diff = 100 - i
        count = 0
        if diff%j==0:
            count = diff//j
        else:
            count = (diff//j)+1
        temp.append(count)
    temp.reverse()
    while len(temp)>0:
        t1 = temp[-1]       # 理쒖긽??
        t2 = 1              # 1媛쒕줈 珥덇린??
        temp.pop()          # ?쒓굅
        while len(temp)!=0 and temp[-1] <= t1:      # 理쒖긽?꾧? ?섎옉 媛숆굅???묒쑝硫?異붽?
            temp.pop()
            t2 += 1
        answer.append(t2)
    return answer

print(solution(	[93, 30, 55], [1, 30, 5]))

File: Programmers/Lv2/Lv2_기능개발.cpp
#include <string>
#include <vector>
#include <queue>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
	vector<int> answer;
	queue<int> q;

	for (int i = 0; i < progresses.size(); i++) {
		int temp = 100 - progresses[i];
		if (temp % speeds[i] != 0)
			temp = temp / speeds[i] + 1;
		else
			temp = temp / speeds[i];
		q.push(temp);
	}

	while (!q.empty()) {
		int temp1 = q.front();
		int temp2 = 1;
		q.pop();
		while (!q.empty() && q.front() <= temp1) {
			q.pop();
			temp2++;
		}
		answer.push_back(temp2);
	}
	return answer;
}